Services/FeatureServices/FeatureServices.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using AutoMapper;
+using ToggleBuddy.API.Helpers;
+using ToggleBuddy.API.Models.Domain;
+using ToggleBuddy.API.Models.DTOs.RequestDTOs;
+using ToggleBuddy.API.Models.DTOs.ResponseDTOs;
+using System.Security.Claims;
+using ToggleBuddy.API.Repositories.Interfaces;
+
+namespace ToggleBuddy.API.Services.FeatureServices
+{
+    public class FeatureServices : IFeatureServices
+    {
+        private readonly IMapper _mapper;
+        private readonly IFeatureRepository _featureRepository;
+        private readonly IProjectRepository _projectRepository;
+        private readonly IUser _user;
+
+        public FeatureServices(
+                IMapper mapper,
+                IFeatureRepository featureRepository,
+                IProjectRepository projectRepository,
+                IUser user
+               )
+        {
+            _mapper = mapper;
+            _featureRepository = featureRepository;
+            _projectRepository = projectRepository;
+            _user = user;
+        }
+
+        public async Task<ApiResponse<FeatureResponseDto>> CreateFeatureAsync(Project project, ClaimsPrincipal userId, FeatureRequestDto featureRequestDto)
+        {
+            var userObject = await _user.GetCurrentUserAsync(userId);
+            if (userObject == null) return new ApiResponse<FeatureResponseDto> { Message = "User not found", Status = ResponseStatus.NotFound };
+
+            var projects = await _projectRepository.GetProjectByIdForCurrentUserAsync(project.Id, userObject.Id);
+            if (projects == null)
+
+                return new ApiResponse<FeatureResponseDto> { Message = "User not found", Status = ResponseStatus.NotFound };
+
+
+            var featureModel = new Feature
+            {
+                Name = featureRequestDto.Name,
+                Description = featureRequestDto.Description,
+                ExpirationDate = featureRequestDto.ExpirationDate,
+                 ProjectId = project.Id
+            };
+
+            await _featureRepository.CreateAsync(featureModel);
+            return new ApiResponse<FeatureResponseDto> { Result = _mapper.Map<FeatureResponseDto>(featureModel), Message = "Feature created successfully", Status = ResponseStatus.Success };
```

Comment:
I'll advice that Services don't return API responses. Sometimes services are used in other services and the return type expected would be a useable datatype to perform further computations. Leave API responses for controllers (which is the interface with the outside world where API responses are useful) and let services just do some application logic work.

Reply:
I guess the naming is wrong here. It actually has nothing to do with the api responses. I'd rename it to ServiceResponse